### PR TITLE
Fix reporting of image scanning to platform

### PR DESCRIPTION
**Pull request by nimrodkor**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Also, allow scanning untagged, locally built images

Example output:
```
       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 2.0.895 
Update available 2.0.895 -> 2.0.900
Run pip3 install -U checkov to update 


sca_image scan results:

Passed checks: 0, Failed checks: 25, Skipped checks: 0

	/Dockerfile (7a0c8a65a226)
	┌────────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────┐
	│ Total CVEs: 25     │ critical: 6        │ high: 11           │ medium: 2          │ low: 6             │ skipped: 0         │
	├────────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────┤
	├────────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────┤
	│ Package            │ CVE ID             │ Severity           │ Current version    │ Fixed version      │ Compliant version  │
	├────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┤
	│ expat              │ CVE-2022-25236     │ critical           │ 2.4.3-r0           │ 2.4.5.post0        │ 2.4.5.post0        │
	│                    │ CVE-2022-23990     │ critical           │                    │ 2.4.4.post0        │                    │
	│                    │ CVE-2022-23852     │ critical           │                    │ 2.4.4.post0        │                    │
	│                    │ CVE-2022-25235     │ high               │                    │ 2.4.5.post0        │                    │
	│                    │ CVE-2022-25315     │ low                │                    │ 2.4.5.post0        │                    │
	│                    │ CVE-2022-25314     │ low                │                    │ 2.4.5.post0        │                    │
	│                    │ CVE-2022-25313     │ low                │                    │ 2.4.5.post0        │                    │
	├────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┤
	│ go                 │ CVE-2021-38297     │ critical           │ 1.16.5             │ 1.16.9             │ 1.17               │
	│                    │ CVE-2022-23806     │ critical           │                    │ 1.16.14            │                    │
	│                    │ CVE-2022-23806     │ critical           │                    │ 1.16.14            │                    │
	│                    │ CVE-2022-23773     │ high               │                    │ 1.16.14            │                    │
	│                    │ CVE-2022-23773     │ high               │                    │ 1.16.14            │                    │
	│                    │ CVE-2022-23772     │ high               │                    │ 1.16.14            │                    │
	│                    │ CVE-2022-23772     │ high               │                    │ 1.16.14            │                    │
	│                    │ CVE-2021-44716     │ high               │                    │ 1.16.12            │                    │
	│                    │ CVE-2021-41772     │ high               │                    │ 1.16.10            │                    │
	│                    │ CVE-2021-41771     │ high               │                    │ 1.16.10            │                    │
	│                    │ CVE-2021-39293     │ high               │                    │ 1.16.8             │                    │
	│                    │ CVE-2021-29923     │ high               │                    │ 1.17               │                    │
	│                    │ CVE-2021-34558     │ medium             │                    │ N/A                │                    │
	│                    │ CVE-2021-36221     │ medium             │                    │ 1.15.15            │                    │
	├────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┤
	│ git                │ CVE-2022-24975     │ high               │ 2.34.1-r0          │ N/A                │ None               │
	├────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────┤
	│ util-linux         │ CVE-2022-0563      │ low                │ 2.37.2-r1          │ 2.37.4.post0       │ 2.37.4.post0       │
	│                    │ CVE-2021-3996      │ low                │                    │ 2.37.3.post0       │                    │
	│                    │ CVE-2021-3995      │ low                │                    │ 2.37.3.post0       │                    │
	└────────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────┘

More details: http://bridgecrew-app-619572639823-nimtest.s3-website-us-west-2.amazonaws.com/projects?repository=nkor93_nkor/images&branch=bc-096340a_master&runId=latest

Process finished with exit code 1
```

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -90,6 +91,7 @@ def __init__(self):
         self.guidelines_api_url = f"{self.api_url}/api/v1/guidelines"
         self.onboarding_url = f"{self.api_url}/api/v1/signup/checkov"
         self.platform_run_config_url = f"{self.api_url}/api/v1/checkov/runConfiguration"
+        self.run_id_url = f"{self.api_url}/api/v1/cicd/data/runs"
```
Comment (tkishel) [inline checkov/common/bridgecrew/platform_integration.py:94]: Defining of URLs has moved to `setup_api_urls()` :)

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -230,13 +233,29 @@ def is_integration_configured(self):
...
 
+    def set_run_details(self) -> None:
+        if self.run_temp_id:
+            request = self.http.request("GET", f'{self.run_id_url}?repositoryId={self.repo_id}&pr={self.run_temp_id}&branch={self.repo_branch}',
+                                        headers=merge_dicts({
+                                            "Authorization": self.get_auth_token(),
+                                            "Content-Type": "application/json"
+                                        }, get_user_agent_header()))
+            response = json.loads(request.data.decode("utf8"))
+            if isinstance(response, list) and len(response) > 0:
```
Comment (gruebel) [inline checkov/common/bridgecrew/platform_integration.py:244]:
```suggestion
            if isinstance(response, list) and response:
```
no need to check the length here

**Review by gruebel — APPROVED** [review]
looks good 👍

**Review by gruebel — APPROVED** [review]
nice, looks good 🚀 
I think the `asyncio` for docker scanning is currently not really needed, because we only do one scan per run, but can leverage this later 😄

**Pull request merged**